### DISCUSSION TITLE
How many waves per level in endless mode?

### DISCUSSION THREAD

**Discussion by ochadenas** [ochadenas/cpudefense#165] [Polls]
Hi all,

as of now, the number of waves in the endless mode equals the level number. On stage 1 there is one wave, on stage 2 there are two ... and on level 420 there are 420 waves.

Well, as I already said earlier, I had never imagined that someone would play more than a few stages, perhaps 15 or 20 or so. But you do, obviously. Which lets me think that these levels will take an enormously long time each one.

So my question is, would you prefer a slower progression of the wave count? For example, proportional to the square root of the level. This would mean that at stage 100, there would not be 10 times as many waves as at stage 10, but only 3. And at stage 400 there would be not 40 times as many waves, but only 6 times as many.

**Comment by Mortis03** [comment]
Please please limit the # of waves in endless mode.  As you said at level 420 there are 420 waves and at an average of 10 seconds a wave, that's 4200 seconds, or 70 mins.  You can typically get all chips maxed out by about level 100, using SHRs, SUBs, and a CLK.  After that its just waiting.  Even on turbo mode it'll take 5 - 10 mins to finish the level.  On turbo mode it also seems like the chips miss attackers they shouldn't miss if was on normal speed.  That might just be my perception, but it certainly seems that way.

Also, it seems like the minimum number of "spaces" for chips is 3 for any giving path to the CPU.  This is fine for lower levels, but at higher levels of 300+, you need a minimum of five "spaces".  One for a CLK chip, and two "sets" of one SHR and one SUB chips.  At level 400+ you pretty much have to restart games until you generate a level with a min of four "spaces" in any given route to the CPU.  

I really enjoy the game, but at 430+ it is that strategy isn't enough and I have pretty much given up at trying to get past level 433.  

Don't get me wrong, it is a great game i have enjoyed and I'm not trying to complain.  But there probably needs to be some tweaking for high levels.  I don't really know of a good solution though.

Thanks for the great game!

**Reply by ochadenas** [reply]
> Also, it seems like the minimum number of "spaces" for chips is 3 for any giving path to the CPU.

Also changed this. Minimum number now is 4 starting from level 200, and 5 for levels above 400.

**Comment by ochadenas** [comment]
Well, there has not been much participation in the survey ... but 100% of the (one) participants think that the number should be reduced.

I'll try the following formula:
$2\sqrt{n}+2$, where $n$ is the stage number.

This means that the first stage starts with 4 waves. At stage 10, we have 8 waves. Then 22 waves at stage 100, 30 at stage 200, 42 at stage 400. And so on. You can calculate the numbers yourselves ...

**Reply by Mortis03** [reply]
Nice. Time to try 433 again
